src/v/raft/consensus.cc

```diff
@@ -908,7 +909,30 @@ ss::future<> consensus::start() {
     vlog(_ctxlog.info, "Starting");
     return _op_lock.with([this] {
         read_voted_for();
-        const bool initial_state = is_initial_state();
+
+        /*
+         * temporary workaround:
+         *
+         * if the group's ntp matches the pattern, then do not load the initial
+         * configuration snapshto from the keyvalue store. more info here:
+         *
+         * https://github.com/vectorizedio/redpanda/issues/1870
+         */
+        const auto& ntp = _log.config().ntp();
+        const auto normalized_ntp = fmt::format(
+          "{}.{}.{}", ntp.ns(), ntp.tp.topic(), ntp.tp.partition());
+        const auto& patterns = config::shard_local_cfg()
+                                 .full_raft_configuration_recovery_pattern();
+        auto initial_state = std::any_of(
+          patterns.cbegin(),
+          patterns.cend(),
+          [&normalized_ntp](const ss::sstring& pattern) {
+              return pattern == "*" || normalized_ntp.starts_with(pattern);
```

Comment:
just confirming that *if* you have a file called `snapshot-1` and by marking this as initial state=true if in the future raft tries to overwrite `snapshot-1` that it would succeed because the code for snapshots overrides ?

Reply:
yeh I think this is ok. in fact, initial=true means that the snapshot will even be removed from the kvstore on bootup. once we remove the flag after Michal fixes it further then it should just go back to normal.